lib/rules-generator.js

```diff
@@ -153,21 +162,28 @@ util.methods(Generator, {
     var thisVar = ast.variable('this');
 
     function registerAsCall(name, methodName) {
-      self.symbols.registerSchema(name, 'Any', undefined, {
+      self.symbols.registerSchema(name, ast.typeType('Any'), undefined, {
         validate: ast.method(['this'], ast.call(ast.reference(ast.cast(thisVar, 'Any'),
                                                               methodName)))
       });
     }
 
-    this.symbols.registerSchema('Any', 'Any', undefined, {
+    this.symbols.registerSchema('Any', ast.typeType('Any'), undefined, {
```

Comment:
When referencing types here would it make sense to have the first two arguments be `Type` values? What you have seems functional, but maybe feels a little off-balance. The prior version of this method used `String` values for the first two args.